extensions/notebook/src/extension.ts

```diff
@@ -53,6 +65,66 @@ async function openNotebook(): Promise<void> {
 	}
 }
 
+async function analyzeNotebook(oeContext?: sqlops.ObjectExplorerContext): Promise<void> {
+	// Ensure we get a unique ID for the notebook. For now we're using a different prefix to the built-in untitled files
+	// to handle this. We should look into improving this in the future
+	let untitledUri = vscode.Uri.parse(`untitled:Notebook-${counter++}`);
+
+	let config = getConfiguration(notebookConfigKey);
+	if (config) {
+		let providerId: string = SQL_NOTEBOOK_PROVIDER;
```

Comment:
As mentioned, let's remove sql check. This is always a PySpark3 scenario since it's inserting spark code